### ISSUE TITLE
Should we compete?

### ISSUE THREAD

**Issue by jtagcat** [openmicroapps/meta#12]
Some apps like Writefreely already exist, should we compete? I'd say no
shorturl.. maybe no as well, there are [many solid](https://github.com/Kickball/awesome-selfhosted#url-shorteners) ones.

**Comment by prologic** [comment]
I agree there are lots of different varieties of everything. But I don't consider what we're doing here "competing" per se. Let's focus on getting the core working, defining the protocols, getting one or two of our own micro apps working before venturing out?

One thing to note as well is; I would not "bring in to the fold" (_i.e: just accept any micro app into the openmicroapps org_). I started writing all this in Go for a very good reason :D 1) to learn of course 2) to build efficient apps -- What do I mean by efficient?

- Single portable binary
- Minimal resource utilisation without writing in C :)
- No web server required or hard to understand, use or configure things.

I can probably think of more... But you get the idea hopefully :) 

That being said however -- We _could_ as part of this issue discuss and open up the possibility of defining a packaging model, a set of standards and features to add to the oma controller (_I renamed the repo_) to support installing other "compatible" micro apps. Docker is really useful here for example.

**Comment by jtagcat** [comment]
> I agree there are lots of different varieties of everything. But I don't consider what we're doing here "competing" per se. Let's focus on getting the core working, defining the protocols, getting one or two of our own micro apps working before venturing out?

Yes please, I guess the first app would be todo? (as as I found yesterday, there isn't much 'competition' for it)

**Comment by prologic** [comment]
Yup let's do todo first.

**Comment by prologic** [comment]
Longer-term I'd like to propose the following requirements:

**Official Open Micro App:**

Backend:

- Must be written in [Go](https://golang.org)
- Must not depend on any C libraries (_no `cgo`_)
- Must implement the defined Protobuf / gRPC protocols (TBD)

Frontend:

- Must use the standard supported OMA CSS and JS libraries

**3rd-party Open Micro Apps:**

Backend:

- Can be written in _any_ language
- Must be packaged as a Docker Image
- Must _at least_ implement the defined REST protocol
- (Optionally) Can implement the Protobuf / gRPC protocol

Frontend:

- Can use any CSS / JS libraries
- (Optionally) use the standard OMA CSS/JS as a different view or the standard view

**Comment by jtagcat** [comment]
FE: Must follow the design requierements/looks.

**Comment by prologic** [comment]
I wonder how much of what we're trying to do here overlaps with [Odoo](https://github.com/odoo)? 🤔

**Comment by jtagcat** [comment]
:rofl: well the difference is that we have micro apps as far as I can see
![image](https://user-images.githubusercontent.com/38327267/62277037-dcf73b80-b434-11e9-87bc-3ad152518ce4.png)
Also they have a more developed app.
